CIP-0054/README.md

```diff
@@ -0,0 +1,227 @@
+
+```
+{
+	"721": {
+		"<policy_id>": {
+			"<asset_name>": {
+				"name": <string>,
+
+				"image": <uri | array>,
+				"mediaType": "image/<mime_sub_type>",
+
+				"description": <string | array>,
+
+				"files": [{
+					"id": <string>
+					"name": <string>,
+					"mediaType": <mime_type>,
+					"src": <uri | array>,
+					<other_properties>
+				}],
+
+				"uses": {
+					"transactions": <string | array>,
+					"tokens": <string | array>,
+					"renderer": <string>	  
+				}
+			}
+		},
+		"version": "1.0"
+	}
+}
+```
+
+Here we have added the “uses” key – any future additions to the Smart NFT API can be implemented by adding additional keys here. 
+We've also added an additional field to the files array - this is to specify a unique identifier to enable the files to be referenced by the Javascript API below. 
+
+### The transactions key
+
+To enable evolving NFTs where the NFT monitors a transaction history and changes in response to new transaction metadata, we define a sub-key called “transactions”, which can contain either a string or an array, specifying the tokens or addresses the NFT wishes to receive the transaction history for. These transaction histories will be provided to the NFT via the Javascript API detailed below.
+
+We also define a special keyword “own” which can be used to monitor the NFT’s own transaction history. So if we wish to create an “evolvable” NFT that can respond to additions to its own transaction history, the “uses” key within the metadata would look like:
+
+```
+	"uses": {
+		"transactions": "own"
+	}
+```
+
+If you wanted to create an evolving NFT which monitors its own transaction history, as well as that of an external smart contract address, the metadata would look like this:
+
+```
+ 	"uses": {
+		"transactions": [
+			"own",
+			"addr1wywukn5q6lxsa5uymffh2esuk8s8fel7a0tna63rdntgrysv0f3ms"
+		]			
+	}
+```
+Finally, we also provide the option to receive the transaction history for a specific token other than the NFT itself (generally this is intended to enable import of Oracle data or control data from an external source – although monitoring an address transaction history could also be used for that). 
+
+When specifying an external token to monitor, you should do so via the token’s fingerprint as in this example:
+
+```
+ 	"uses": {
+		"transactions": [
+			"own",			
+			"asset1frc5wn889lcmj5y943mcn7tl8psk98mc480v3j"
+		]
+	}
+```
+
+### The tokens key
+
+To enable modifier tokens - (that is, a token which you can hold alongside a Smart NFT which changes the Smart NFT's appearance or behaviour in some way); we provide a way for the Smart NFT to monitor the tokens held by a specific address. Similarly to the “transactions” key, the “tokens” key will also accept either a string or array.
+
+In this case we define the “own” keyword to mean the tokens held by the same stake key that currently holds the Smart NFT itself.
```

Comment:
What about stake part not directly associated to a single stake key? For instance, a stake part captured by a native monetary script as a combination of three keys? Or simply, an address without stake part? Would those be simply excluded from the 'own' keyword?